PyU4V/replication.py

```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+
+        This function should be run on initial SRDF configuration once zoning
+        or IP routing is configured, prior to first RDF group being configured.
+
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param port_id: port number -- int
+        :param array_id: 12 digit serial number for Source  -- str
+        :returns: remote port details --dict
+        """
+
+        if not array_id:
+            array_id = self.array_id
+        return self.get_resource(
+            category=REPLICATION,
+            resource_level=SYMMETRIX, resource_level_id=array_id,
+            resource_type=RDF_DIRECTOR, resource_type_id=director_id,
+            resource=PORT, resource_id=port_id, object_type=REMOTE_PORT)
+
+    def create_rdf_group(self, local_director_port_list, remote_array_id,
+                         label, local_rdfg_number, remote_rdfg_number,
+                         remote_director_port_list, array_id=None):
+        """Create a new RDF group between directors and ports on two PowerMax.
+
+        If this is the first connection between 2 arrays please ensure that you
+        run get_rdf_remote_port_details and create the initial connection
+        group using only the first source port you can extract details for one
+        or more remote ports for the desired target array and feed into this
+        function. Additional Ports can be added with modify_rdf_group function.
+
+        :param local_director_Port_list: list of local directors and ports for
+                                        group e.g [RF-1E:1, RF-2E:1] -- list
+        :param remote_array_id: 12 digit serial number of remote array  -- str
+        :param label: Label for group up to 10 characters -- str
+        :param local_rdfg_number: rdfg for the local array-- int
+        :param remote_rdfg_number rdfg for the remote array -- int
+        :param remote_director_port_list: list of remote directors and ports to
+                                          group e.g [RF-1E:1, RF-2E:1] -- list
+        :param array_id: array_id: 12 digit serial number of Source (R1) array,
+                        if no array is specified the array in config file or
+                        api connection will be default -- str
+        """
+        if not array_id:
+            array_id = self.array_id
+        local_ports = []
+        remote_ports = []
+        for port in local_director_port_list:
+            dir_id = port.split(':')[0]
+            port_no = port.split(':')[1]
+            local_ports.append({'symmetrixID': array_id,
+                                'directorId': dir_id,
+                                'portNumber': port_no})
+        for port in remote_director_port_list:
+            dir_id = port.split(':')[0]
+            port_no = port.split(':')[1]
+            remote_ports.append({'symmetrixID': remote_array_id,
+                                 'directorId': dir_id, 'portNumber': port_no})
```

Comment:
Nit: It would be good to keep symmetrixID and similar keys to constants, saves us in the future so we only need to change constants in one place if they change